homeworks/dmytro.kondaurov_DmitriyKondaurov/3-async/main.js

```diff
@@ -0,0 +1,82 @@
+const fieldForPosts = document.body.querySelector('[data-main-box]');
+let listOfPosts = [];
+const postFilter = document.body.querySelector('[data-filter]');
+const postSort = document.body.querySelector('[data-sort]');
+let template = document.querySelector('[data-post-template]');
+
+const hideLoaderGifImage = () => {
+    document.querySelector('[data-main-content-loader]').classList.add('hide');
+};
+
+const addPost = (title, text) => {
+    const newEl = template.content.querySelector('[data-post-item]');
+    const cloneNewEl = newEl.cloneNode(true);
+    cloneNewEl.querySelector('[data-post-text]').textContent = text;
+    cloneNewEl.querySelector('[data-post-title]').textContent = title;
+    fieldForPosts.append(cloneNewEl);
+};
+
+const filterEl = (post) => post.querySelector('[data-post-title]').textContent.localeCompare(postFilter.value) === 1;
+
+const sortElForward = (a, b) => {
+    const aTitle = a.querySelector('[data-post-title]').textContent.toUpperCase();
+    const bTitle = b.querySelector('[data-post-title]').textContent.toUpperCase();
+    if (aTitle < bTitle) {
+        return -1;
+    }
+    if (aTitle > bTitle) {
+        return 1;
+    }
+    return 0;
+};
+
+const sortElReverse = (a, b) => {
+    const aTitle = a.querySelector('[data-post-title]').textContent.toUpperCase();
+    const bTitle = b.querySelector('[data-post-title]').textContent.toUpperCase();
+    if (aTitle < bTitle) {
+        return 1;
+    }
+    if (aTitle > bTitle) {
+        return -1;
+    }
+    return 0;
+};
+
+const reRenderPosts = () => {
+    fieldForPosts.innerHTML = '';
+    listOfPosts
+        .filter(filterEl)
+        .sort((a, b) => {
+            if (postSort.value === 'a-z') {
+                return sortElForward(a, b);
+            } if (postSort.value === 'z-a') {
+                return sortElReverse(a, b);
+            } return 0;
```

Comment:
Тут трошки не так щось з переносами рядків
```suggestion
            if (postSort.value === 'a-z') {
                return sortElForward(a, b);
            }
            if (postSort.value === 'z-a') {
                return sortElReverse(a, b);
            }
            return 0;
```

Reply:
done